ext/StructArraysAdaptExt.jl

```diff
@@ -0,0 +1,16 @@
+module StructArraysAdaptExt
+# Use Adapt allows for automatic conversion of CPU to GPU StructArrays
+using Adapt, StructArrays
+@static if !applicable(Adapt.adapt, Int)
+    # Adapt.jl has curried support, implement it ourself
+    adpat(to) = Base.Fix1(Adapt.adapt, to)
+    if VERSION < v"1.9.0-DEV.857"
+        @eval function adapt(to::Type{T}) where {T}
+            (@isdefined T) || return Base.Fix1(Adapt.adapt, to)
+            AT = Base.Fix1{typeof(Adapt.adapt),Type{T}}
+            return $(Expr(:new, :AT, :(Adapt.adapt), :to))
+        end
+    end
+end
+Adapt.adapt_structure(to, s::StructArray) = replace_storage(adapt(to), s)
```

Comment:
```suggestion
Adapt.adapt_structure(to, s::StructArray) = replace_storage(Base.Fix1(Adapt.adapt, to), s)
```

Can't we just do this and remove all the code above? Why do we need all of the above code to create a curried version that we just use here?

Reply:
`Base.Fix1` has stability issue before julia 1.9 for Type input. I just copy the patch from Adapt.jl.